common/insightagent-boilerplate.py

```diff
@@ -0,0 +1,587 @@
+#!/usr/bin/env python
+import ConfigParser
+import collections
+import json
+import logging
+import os
+import re
+import socket
+import sys
+import time
+import pytz
+from optparse import OptionParser
+from itertools import islice
+from datetime import datetime
+import dateutil
+from dateutil.tz import tzlocal
+import urlparse
+import httplib
+import requests
+
+'''
+This script gathers data to send to Insightfinder
+'''
+
+
+def start_data_processing():
+    """ TODO: replace with your code.
```

Comment:
This could be a good place to allow a modular processing step (check for an ancillary parser file and use to process the data if present)

Reply:
Agreed, but due to the current design (copy it and edit from there) vs an importable module, it would be awkward to change where the agent config vars are grabbed. I want to get to an importable module where we could be more thoughtful about the design from the get-go; this was 90% written while re-writing things for the Zipkin agent, so it was just restructuring and formalizing what we were doing for agents before.

In short, yes, but IMO it's out of scope.